### DISCUSSION TITLE
Getting same value in props.value for different checkboxes

### DISCUSSION THREAD

**Discussion by azad-steadybiz** [react-hook-form/react-hook-form#2800] [Q&A]
I am saving all the values of checkbox in array, but for view screen I want to populate the checkboxes on the basis of response from backend.
I am trying to reset the value of a checkbox variable on useEffect, but only the first checkbox is populated rest are not populating even if their value present in the response.

I found out, the first value of checkbox is repeating in props.value in render function. Please help

CSB: https://codesandbox.io/s/dazzling-cray-en573?file=/src/App.js

**Comment by bluebill1049** [comment]
here the steps you probably should take:

1. learn Controller API for the external controlled component: https://react-hook-form.com/api#Controller
2. take a look at this issue for solutions: https://github.com/react-hook-form/react-hook-form/issues/1517

**Comment by azad-steadybiz** [comment]
I've gone through both of the points. It will be helpful for me if you can point out the mistake.

**Reply by bluebill1049** [reply]
you should use Controller, which is what I meant for point 1. there is a codesandbox in that issue: https://codesandbox.io/s/material-demo-bzj4i?file=/demo.js

**Reply by bluebill1049** [reply]
hope those are helpful :)

**Reply by AzadAnsari35** [reply]
I am using controller, please go through the sandbox.
